src/Uno.UI/Microsoft/UI/Xaml/Controls/TreeView/TreeViewList.cs

```diff
@@ -349,7 +349,17 @@ protected override void PrepareContainerForItemOverride(DependencyObject element
 			{
 				DispatcherQueue.TryEnqueue(() =>
 				{
-					itemNode.IsExpanded = itemContainer.IsExpanded;
+					// Uno Specific: this is actually a bug in WinUI, copy WinUI's fix when https://github.com/microsoft/microsoft-ui-xaml/issues/9549 is resolved
+					// itemNode.IsExpanded = itemContainer.IsExpanded;
+					var bindingExists = itemContainer.GetBindingExpression(TreeViewItem.IsExpandedProperty) is { };
+					if (bindingExists)
+					{
+						itemNode.IsExpanded = itemContainer.IsExpanded;
+					}
+					else
+					{
+						itemContainer.IsExpanded = itemNode.IsExpanded;
+					}
```

Comment:
use a setup like this when applying patch to microsoft(tm) code
```suggestion
#if HAS_UNO // Uno Specific: this is actually a bug in WinUI, copy WinUI's fix when https://github.com/microsoft/microsoft-ui-xaml/issues/9549 is resolved
					itemNode.IsExpanded = itemContainer.IsExpanded;
#else
					// todo: might want to explain your logic here a bit
					if (itemContainer.GetBindingExpression(TreeViewItem.IsExpandedProperty) is { })
					{
						itemNode.IsExpanded = itemContainer.IsExpanded;
					}
					else
					{
						itemContainer.IsExpanded = itemNode.IsExpanded;
					}
#endif
```